pkg/service/credential/storage/bolt.go

```diff
@@ -1,35 +1,138 @@
 package storage
 
 import (
+	"crypto/rand"
 	"fmt"
-	"strings"
-
 	"github.com/TBD54566975/ssi-sdk/credential/signing"
 	"github.com/goccy/go-json"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"math/big"
+	"strings"
 
 	"github.com/tbd54566975/ssi-service/internal/util"
 	"github.com/tbd54566975/ssi-service/pkg/storage"
 )
 
 const (
-	namespace                = "credential"
+	credentialNamespace           = "credential"
+	statusListCredentialNamespace = "status-list-credential"
+	statusListIndexNamespace      = "status-list-index"
+
 	credentialNotFoundErrMsg = "credential not found"
 )
 
 type BoltCredentialStorage struct {
 	db *storage.BoltDB
 }
 
+type StatusListIndex struct {
+	Index int `json:"index"`
+}
+
 func NewBoltCredentialStorage(db *storage.BoltDB) (*BoltCredentialStorage, error) {
 	if db == nil {
 		return nil, errors.New("bolt db reference is nil")
 	}
+
+	// TODO: (Neal) there is a current bug with our Bolt implementation where if we do a GET without anything in the db it will throw an error
+	// Doing initial writes and then deleting will "warm up" our database and when we do a GET after that it will not crash and return empty list
```

Comment:
This is incredibly suspect and I strongly suspect there is another error going around.

Reply:
it's because the namespace bolt is using is created on first write. we should open up another issue to address it.
solutions:

1. seed the DB with a first entry 
2. detect the namespace not exists error and throw another error

Reply:
issue created - https://github.com/TBD54566975/ssi-service/issues/176